spec/app_support/reservations/date_support_spec.rb

```diff
@@ -0,0 +1,271 @@
+
+    def product
+      OpenStruct.new(schedule_rules: [])
+    end
+
+    def restrict_purchase?
+      false
+    end
+
+    def usage_rate
+      1
+    end
+
+    def usage_subsidy
+      0
+    end
+  end
+
+  subject(:reservation) do
+    build_stubbed(
+      :reservation,
+      reserve_start_at: reserve_start_at,
+      reserve_end_at: reserve_end_at,
+      actual_start_at: actual_start_at,
+      actual_end_at: actual_end_at,
+    )
+  end
+
+  let(:reserve_start_at) { nil }
+  let(:reserve_end_at) { nil }
+  let(:actual_start_at) { nil }
+  let(:actual_end_at) { nil }
+
+  describe "#actual_duration_mins" do
+    context "when @actual_duration_mins is set" do
+      before { reservation.actual_duration_mins = minutes }
+
+      context "to a string representation of an integer" do
+        let(:minutes) { "5" }
+
+        it "converts @actual_duration_mins to an integer, as-is" do
+          expect(reservation.actual_duration_mins).to eq(5)
+        end
+      end
+
+      context "to a string representation of a float" do
+        context "where the difference is < 0.5" do
+          let(:minutes) { "0.25" }
+
+          it "has no minimum and rounds down to 0" do
+            expect(reservation.actual_duration_mins).to eq(0)
+          end
+        end
+
+        context "where the fraction is < 0.5" do
+          let(:minutes) { "1.25" }
+
+          it "truncates the fraction" do
+            expect(reservation.actual_duration_mins).to eq(1)
+          end
+        end
+
+        context "where the fraction is 0.5" do
+          let(:minutes) { "1.5" }
+
+          it "truncates the fraction" do
+            expect(reservation.actual_duration_mins).to eq(1)
+          end
+        end
+
+        context "where the fraction is > 0.5" do
+          let(:minutes) { "1.99" }
+
+          it "truncates the fraction" do
+            expect(reservation.actual_duration_mins).to eq(1)
+          end
+        end
+      end
+    end
+
+    context "when @actual_duration_mins is unset" do
+      context "and actual_start_at is set to 00:00:59" do
+        let(:actual_start_at) { Time.zone.parse("2015-01-01T00:00:59") }
+
+        [
+          ["2015-01-01T00:01:01", 1],
+          ["2015-01-01T00:01:59", 1],
+          ["2015-01-01T00:02:01", 2],
+          ["2015-01-01T00:02:59", 2],
+          ["2015-01-01T01:00:00", 60],
+          ["2015-01-01T01:00:59", 60],
+        ]
+        .each do |timestring, expected_minutes|
+          context "and the actual_end_at is #{timestring}" do
+            let(:actual_end_at) { Time.zone.parse(timestring) }
+            let(:price_policy_calculation) do
+              DummyPricePolicy.new.estimate_cost_and_subsidy(actual_start_at, actual_end_at)
+            end
+
+            it "returns #{expected_minutes}" do
+              expect(reservation.actual_duration_mins).to eq(expected_minutes)
+            end
+
+            it "matches InstrumentPricePolicyCalculations' cost at $1 per minute" do
```

Comment:
I think it was a smart idea to write these tests to make sure we're matching the price calculations. I'm not sure it's worth keeping those tests around, though. It clutters up this test, and I'm not sure it'll be clear on down the road why these tests exist.